qutebrowser/browser/hints.py

```diff
@@ -199,6 +210,12 @@ def _hint_strings(self, elems):
         Return:
             A list of hint strings, in the same order as the elements.
         """
+        if config.get('hints', 'mode') == 'words':
+            try:
+                return self._word_hinter.hint(elems)
+            except WordHintingError as e:
+                message.error(self._win_id, str(e), immediately=True)
+                # falls back on letter hints
         if config.get('hints', 'mode') == 'number':
```

Comment:
please change this to `elif` to make it more clear those are mutually exclusive. It might also make sense to do `hint_mode = config.get('hints', 'mode')` somewhere above the if-block.